module/zcommon/zfeature_common.c

```diff
@@ -430,6 +430,11 @@ zpool_feature_init(void)
 	    "Support for separate allocation classes.",
 	    ZFEATURE_FLAG_READONLY_COMPAT, NULL);
 	}
+
+	zfeature_register(SPA_FEATURE_RESILVER_DEFER,
+	    "com.datto:resilver_defer", "resilver_defer",
+	    "Support for defering new resilvers when one is already running.",
+	    ZFEATURE_FLAG_READONLY_COMPAT, NULL);
```

Comment:
what do you mean that it can't be disabled once it is enabled? Cant you just disable it like any other feature?

Reply:
Keep in mind the distinction between `disabled` (not in use, not allowed to use), `enabled` (but inactive - not in use, but allowed to use), and `active` (in use, downgrade to older software not supported).  There is no feature disabling in ZFS.  However features can go back and forth between `enabled` and `active`.

All that said, I disagree with @behlendorf's suggestion of making this a pool property, because I think that the new behavior is appropriate in nearly all cases.

Reply:
> There is no feature disabling in ZFS

This is specifically my concern.  I agree the new behavior may be appropriate for the majority of cases, but there are operational reasons for preferring the existing behavior.  For example, any existing infrastructure built on ZFS may depend on it.

This change provides no mechanism to opt-out except at pool creation time.  By the time you discover this is an issue, it could be too late and it's non-reversible.  That's not a good user experience.

If it were possible to run `zfs set feature@resilver_defer=disabled <pool>` when the feature is `enabled` (but not `active`) that would be fine.  Adding a module option would also be reasonable. 
 As long as there's a mechanism if you absolutely need it.

Reply:
@behlendorf I think you can accomplish what you want by using `zpool create -o feature@resilver_defer=disabled` <poolname> <devices>`

Reply:
I think @behlendorf was saying there is no way to disable a feature after the feature is enabled.